src/main/java/org/codeforamerica/messaging/models/Message.java

```diff
@@ -31,8 +32,8 @@ public class Message {
     String toPhone;
     @Email
     String toEmail;
-    @NotBlank
-    String body;
+    String emailBody;
```

Comment:
It is not part of this story, but I really think we should remove `body` and `subject` from `Message`. With the current implementation, we will have the fields show up in 3-4 places in a `GET messages/{id}.

Reply:
Completely agree. One snag is that the `EmailMessage` and `SmsMessage` don't get created when the message is scheduled, only when it is sent, so when the templates are processed after the initial request, the values are stored on the `Message`. I think ideally the Email & Sms Message should get created and scheduled and stored in the DB at the same time as the Message.

Reply:
That shouldn't be much of a snag as the associated `TemplateVariant` will provide some information about the intended messages.

Reply:
I was thinking it would be nice to process the data from the `TemplateVariant` at the time the request is processed rather than waiting until it is sent, and store the output at that time if it is valid. Allows the response to say "x number of rows could not be processed".

Reply:
Although I suppose the template could be re-processed at sending time, that just seems awkward to me.

Reply:
I played around with my idea for this, but the new concern I would have is creating `SmsMessage` and `EmailMessage` before we know the `providerMessageId`. I'm wondering if we can find a way to work around that, because overall I like having "scheduled" entries in the database for the specific messages that will be sent.

Reply:
I thought about this more, and even though re-processing the `templateVariant` at sending time seems awkward to me, it's a cheap operation. It would also mean if a large batch is created and then cancelled then all `Message`, `SmsMessage`, and `EmailMessage` would have to be updated. 

On this same thread, I was wondering if you can ever envision a scenario where a single message within a scheduled batch would need to be edited (cancelled/rescheduled) before it was sent? I don't think so, but we should probably make it clear that batches are "all or nothing" once created.